packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -438,17 +429,19 @@ export default class ApplicationShell extends React.Component {
                           ),
                         }),
                   };
-                  // TODO: when we remove the login routes from the AppShell, this component
-                  // should do a redirect/hard page reload to the login route.
-                  // For development, we should only do a redirect to the staging domain, as the
-                  // login screen will not be part of the AppShell anymore.
-                  return (
-                    <Redirect
-                      to={{
-                        pathname: '/login',
-                        query: searchQuery,
-                      }}
-                    />
+
+                  /**
+                   * NOTE:
+                   *   Given the application runs in production we redirect to that host's login application.
+                   *   Given the application is in development mode (locally) we redirect to the application on staging.
+                   */
+                  const frontendHost =
+                    process.env.NODE_ENV === 'production'
```

Comment:
We should use `servedByProxy` to determine if we're running locally or on a production environment